### ISSUE TITLE
wp scaffold and WPCS

### ISSUE THREAD

**Issue by sandrodz** [wp-cli/scaffold-command#109]
Files generated by wp scaffold command are not WordPress coding standard compliant!

For example cpt generated:

<img width="1280" alt="screen shot 2017-12-22 at 3 39 29 pm" src="https://user-images.githubusercontent.com/8479569/34301798-5635d66c-e72e-11e7-9062-378b07b4a648.png">

https://github.com/wp-cli/wp-cli/issues/4589

**Comment by gitlost** [comment]
Close via https://github.com/wp-cli/scaffold-command/pull/110

**Comment by gitlost** [comment]
Oops, maybe the other files apart from CPT aren't compliant?

**Comment by gziolo** [comment]
@schlessera, @danielbachhuber - is there any Behat logic that would run linters on generated files? That would be great to ensure we scaffold files that are WordPress compliant.

**Comment by schlessera** [comment]
@gziolo No, but that would be a good addition. I'll create an issue for that.

**Comment by gziolo** [comment]
Awesome, thanks 🙇

**Comment by thrijith** [comment]
to keep track what else needs work
- [ ] wp scaffold _s
- [ ] wp scaffold block
- [ ] wp scaffold child-theme
- [ ] wp scaffold plugin
- [ ] wp scaffold plugin-tests
- [ ] wp scaffold post-type
- [ ] wp scaffold taxonomy
- [ ] wp scaffold theme-tests

**Comment by gitlost** [comment]
Good idea @thrijith !

**Comment by gitlost** [comment]
Going to remove this from the 1.1.12 milestone as incomplete at this late stage...

**Comment by gziolo** [comment]
> @gziolo you fixed scaffold block right?

Yes, I fixed using Gutenberg's configuration. For PHP, it should be the same as core uses. JS part follows Eslint setup, but I hope it is similar to what JSHint promotes.

**Comment by NicktheGeek** [comment]
I just checked plugin and plugin test scaffold. Both are coming up good to go for wpcs so those two can/should be checked off on the list above. I'll check the _s next.

**Comment by NicktheGeek** [comment]
Looks like `plugin` and `plugin-tests` are mostly good to go. I did find an error generated on bootstrap.php. It is a false positive so I added a PR that uses a wpcs exclusion on the line to suppress the error.

**Comment by ernilambar** [comment]
Command `wp scaffold post-type movie --label=Movie --theme=sample-theme`

I used rules from the file we are distributing in scaffold - https://github.com/wp-cli/scaffold-command/blob/main/templates/.phpcs.xml.dist

![Screenshot 2024-06-11 at 11 39 05 AM](https://github.com/wp-cli/scaffold-command/assets/2098823/237a5287-07fd-4112-a9ed-d879ad1f40c0)

Is it worth revisiting our scaffold snippets as in 6 years there has been lots of changes in the sniffer?

CC @wp-cli/committers

**Comment by swissspidy** [comment]
Seems reasonable to revisit this & implement some tests to ensure that the scaffolded code always passes PHPCS.
